homeassistant/components/google_generative_ai_conversation/conversation.py

```diff
@@ -156,18 +158,32 @@ async def async_process(
                 )
             tools = [_format_tool(tool) for tool in llm_api.async_get_tools()]
 
-        raw_prompt = self.entry.options.get(CONF_PROMPT, DEFAULT_PROMPT)
+        if self.entry.options.get(CONF_RECOMMENDED, False):
```

Comment:
Would it be better to simplify the code here by clearing the options in the config flow if you select recommended? That might also be better since if a user changes settings, then enables recommended, and then disables recommended they would expect the settings to have gone back to the recommended ones instead of remembering the previous ones.